cairo_zero/kakarot/precompiles/kakarot_precompiles.cairo

```diff
@@ -139,3 +152,161 @@ namespace KakarotPrecompiles {
         return (0, output, CAIRO_MESSAGE_GAS, FALSE);
     }
 }
+
+namespace Internals {
+    // @notice Takes an array of cairo calls and executes them.
+    // @dev If any of the internal calls revert, the entire batch reverts.
+    // @param caller_address The address of the caller of the precompile.
+    // @param calls_len The length of the calls array.
+    // @param calls The calls to execute.
+    // @returns reverted Errors.EXCEPTIONAL_HALT if reverted, FALSE otherwise.
+    func execute_multiple_cairo_calls{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(caller_address: felt, call_selector: felt, calls_len: felt, calls: felt*) -> (
+        output_len: felt, output: felt*, reverted: felt
+    ) {
+        alloc_locals;
+
+        if (calls_len == 0) {
+            let (output) = alloc();
+            return (0, output, FALSE);
+        }
+
+        // Ensure that the current remaining calls_len  >= MIN_EVM_ENCODED_STARKNET_CALL_BYTES
+        // Otherwise the input is malformed
+        let is_input_invalid = is_nn(MIN_EVM_ENCODED_STARKNET_CALL_BYTES - (calls_len + 1));
+        if (is_input_invalid != 0) {
+            let (revert_reason_len, revert_reason) = Errors.outOfBoundsRead();
+            return (revert_reason_len, revert_reason, Errors.EXCEPTIONAL_HALT);
+        }
+
+        // Input is formatted as an array of:
+        // [starknet_address: bytes32][starknet_selector:bytes32][data_offset: bytes32][data_len: bytes32][data: bytes[]]]
+
+        // Load address and cairo selector called
+        // Unsafe to assume that the 32 bytes in input do not overflow a felt - responsibility of the caller to format its input correctly
+        let to_starknet_address = Helpers.bytes32_to_felt(calls);
```

Comment:
transform to Uint256 and compare to Prime to ensure this does not overflow ?

Reply:
bytes32_to_uint256 -> ensure not > PRIME -> reconstruct felt from (low, high)? 

idk if we should put guard rails around this. @ClementWalter thoughts?